lexibank_tryonsolomon.py

```diff
@@ -35,26 +34,33 @@ class Dataset(BaseDataset):
     id = "tryonsolomon"
     language_class = Language
 
-    def cmd_download(self, **kw):
+    form_spec = FormSpec(
+        brackets={"(": ")"},
+        separators=";/,",
+        missing_data=('?', '-', '', ''),
+        strip_inside_brackets=True
+    )
+    
+    def cmd_download(self, args):
         pass
 
-    def cmd_install(self, **kw):
-        source = self.raw.read_bib()[0]
+    def cmd_makecldf(self, args):
+        source = self.raw_dir.read_bib()[0]
 
-        conn = sqlite3.connect(self.raw.posix("tryon.db"))
+        conn = sqlite3.connect(self.raw_dir / "tryon.db")
         cursor = conn.cursor()
         cursor.execute(QUERY)
 
-        with self.cldf as ds:
-            ds.add_sources(source)
-            ds.add_concepts(id_factory=lambda c: slug(c.label))
-            ds.add_languages(id_factory=lambda l: slug(l["Name"]))
-            for i, (lang, param, value) in enumerate(cursor.fetchall(), 1):
-                if value:
-                    ds.add_lexemes(
-                        Language_ID=slug(lang),
-                        Parameter_ID=slug(param),
-                        Value=self.lexemes.get(value, value),
-                        Source=[source.id],
-                    )
+        args.writer.add_sources(source)
```

Comment:
it's enough two write `args.writer.add_sources()`, it automatically searches for `sources.bib` in your `raw` folder.

Reply:
thanks Mattis - made all the requested changes in the next commit..